Revamp integration,streaming quick start guides (#4592)
wso2/docs-apim: en/docs/get-started/integration-quick-start-guide.md

Let's get started with WSO2 Micro Integrator by running a simple integration use case in your local environment. 

### Before you begin...

1. Install [Oracle Java SE Development Kit (JDK)](http://java.sun.com/javase/downloads/index.jsp) version 11 or 1.8 and set the `JAVA_HOME` environment variable.
   For more information on setting the `JAVA_HOME` environment variable for different operating systems, see [Setup and Install]({{base_path}}/install-and-setup/install/installing-the-product/installing-api-m-runtime/).
2. Go to the [WSO2 Micro Integrator web page](https://wso2.com/integration/micro-integrator/#), click **Download**, and then click **Zip Archive** to download the Micro Integrator distribution as a ZIP file.

> 3. Optionally, Go to the [API Manager Tooling web page](https://wso2.com/api-management/tooling/), and download WSO2 Integration Studio.

Reviewer: ```suggestion
3. Optionally, navigate to the [API Manager Tooling web page](https://wso2.com/api-management/tooling/), and download WSO2 Integration Studio.
```

Reviewer: Why is this optional? Don't you need Integration Studio for this?

Author: Integration Studio is needed to create the Integration. But in QSG we have provided an already created and tested solution. IS is needed only If one needs to observe the project.

    !!! Info
        For more information, see the [installation instructions]({{base_path}}/install-and-setup/install-and-setup-overview/#installing_1).

4. Download the [sample files]({{base_path}}/assets/attachments/quick-start-guide/mi-qsg-home.zip). From this point onwards, let's refer to this folder as `<mi-qsg-home>`.
5. Download [curl](https://curl.haxx.se/) or a similar tool that can call an HTTP endpoint.

### What you'll build

This is a simple service orchestration scenario. The scenario is about a basic healthcare system where the Micro Integrator is used to integrate two back-end hospital services to provide information to the client.

Most healthcare centers have a system that is used to make doctor appointments. To check the availability of the doctors for a particular time, users typically need to visit the hospitals or use each and every online system that is dedicated to a particular healthcare center. Here, we are making it easier for patients by orchestrating those isolated systems for each healthcare provider and exposing a single interface to the users.

![Scenario]({{base_path}}/assets/img/integrate/quick-start-guide/mi-quick-start-guide.png)

    !!! Tip

>         You may export <mi-qsg-home>/HealthcareIntegrationProject to Integration Studio to view the project structure.

Reviewer: ```suggestion
        You may export `<mi-qsg-home>/HealthcareIntegrationProject` to Integration Studio to view the project structure.
```

Author: fixed

Let’s implement a simple integration solution that can be used to query the availability of doctors for a particular category from all the available healthcare centers.

 
### Step 1 - Set up the workspace

To set up the integration workspace for this quick start guide, we will use an integration project that was built using WSO2 Integration Studio:

> 1. Go to the `<mi-qsg-home>` directory. 

Reviewer: ```suggestion
1. Navigate to the `<mi-qsg-home>` directory. 
```

Author: fixed

Follow the steps given below to run the integration artifacts we developed on a Micro Integrator instance that is installed on a VM.

1. Start the back-end services.

    Two mock hospital information services are available in the `DoctorInfo.jar` file located in the `<mi-qsg-home>/Backend/` directory. 
    Open a terminal window, navigate to the `<mi-qsg-home>/Backend/` folder and use the following command to start the services:
    
    ```bash
    java -jar DoctorInfo.jar
    ```
   

2. Deploy the Healthcare service

    Copy the CAR file of the Healthcare service (HealthcareIntegrationProjectCompositeExporter_1.0.0-SNAPSHOT.car) from the `<mi-qsg-home>/HealthcareIntegrationProject/HealthcareIntegrationProjectCompositeExporter/target/` directory to the `<MI_HOME>/repository/deployment/server/carbonapps` directory.

3. Start the Micro Integrator

    Follow the steps relevant to your OS as shown below.

    1. Open a terminal and navigate to the `<MI_HOME>/bin` folder.
    2. Execute the relevant command:

        ```bash tab= 'On MacOS/Linux/CentOS'
        sh micro-integrator.sh
        ```

        ```bash tab= 'On Windows'
        micro-integrator.bat
        ```

> 4. (Optional) Start the Dashboard

Reviewer: ```suggestion
4. (Optional) Start the Dashboard.
```

Author: fixed

    If you want to view the integration artifacts deployed in the Micro Integrator, you can start the dashboard. The instructions on running the MI dashboard is given in the installation guide:

    1.  [Install]({{base_path}}/install-and-setup/install/installing-the-product/installing-mi-dashboard) the MI dashboard.
    2.  [Start]({{base_path}}/install-and-setup/install/installing-the-product/running-the-mi-dashboard) the MI dashboard.
    

> 5. Invoke the healthcare service

Reviewer: ```suggestion
5. Invoke the healthcare service.
```

Author: fixed

- **Backend**: This contains an executable .jar file that contains mock back-end service implementations for the Pine Valley Hospital and Grand Oak Hospital.

### Step 2 - Running the integration artifacts

Follow the steps given below to run the integration artifacts we developed on a Micro Integrator instance that is installed on a VM.

1. Start the back-end services.

    Two mock hospital information services are available in the `DoctorInfo.jar` file located in the `<mi-qsg-home>/Backend/` directory. 
    Open a terminal window, navigate to the `<mi-qsg-home>/Backend/` folder and use the following command to start the services:
    
    ```bash
    java -jar DoctorInfo.jar
    ```
   

> 2. Deploy the Healthcare service

Reviewer: ```suggestion
2. Deploy the Healthcare service.
```

    Copy the CAR file of the Healthcare service (HealthcareIntegrationProjectCompositeExporter_1.0.0-SNAPSHOT.car) from the `<mi-qsg-home>/HealthcareIntegrationProject/HealthcareIntegrationProjectCompositeExporter/target/` directory to the `<MI_HOME>/repository/deployment/server/carbonapps` directory.

> 3. Start the Micro Integrator

Reviewer: ```suggestion
3. Start the Micro Integrator.
```

    1.  [Install]({{base_path}}/install-and-setup/install/installing-the-product/installing-mi-dashboard) the MI dashboard.
    2.  [Start]({{base_path}}/install-and-setup/install/installing-the-product/running-the-mi-dashboard) the MI dashboard.
    

5. Invoke the healthcare service.

    Open a terminal and execute the following curl command to invoke the service:

    ```bash
    curl -v http://localhost:8290/healthcare/doctor/Ophthalmologist
    ```

    Upon invocation, you should be able to observe the following response:

    ```bash
    [ 
       [ 
          { 
             "name":"John Mathew",
             "time":"03:30 PM",
             "hospital":"Grand Oak"
          },
          { 
             "name":"Allan Silvester",
             "time":"04:30 PM",
             "hospital":"Grand Oak"
          }
       ],
       [ 
          { 
             "name":"John Mathew",
             "time":"07:30 AM",
             "hospital":"pineValley"
          },
          { 
             "name":"Roma Katherine",
             "time":"04:30 PM",
             "hospital":"pineValley"
          }
       ]
    ]
    ```
    Congratulation!

>     Now, you have created a your first integration service. You may follow steps given below to expose the service as a Managed API in API Manager.

Reviewer: ```suggestion
    Now you have created your first integration service. You may follow the steps given below to expose the service as a Managed API in API Manager.
```

    
## Exposing an Integration Service as a Managed API

The REST API you deployed in the Micro Integrator is an **integration service** for the API Manager. Now, let's look at how you can expose the integration service to the API Management layer and generate a managed API by using the service.

> ### Step 1 - Expose as a service 

Reviewer: ```suggestion
### Step 1 - Expose your integration as a service 
```

1. Start the API Manager runtime:

    1.  Go to the [WSO2 API Manager website](https://wso2.com/api-management/), click **TRY IT NOW**, and then click **Zip Archive** to download the API Manager distribution as a ZIP file.
    2.  Extract the ZIP file.
    3.  Start WSO2 API Manager:
    

>         Open a terminal, navigate to the `<API-M_HOME>/bin` folder and execute the relevant command. 

Reviewer: ```suggestion
        Open a terminal, navigate to the `<API-M_HOME>/bin` folder, and execute the relevant command. 
```

        
    3. See that the `HealthcareAPI` is listed as a service.

### Step 2 - Create a managed API using the Integration Service

1.  Click on the `HealthcareAPI` that is in the service catalog.

2.  Click **Create API**.

     This opens the **Create API** dialog box with the API details that are generated based on the service.

    <a href="{{base_path}}/assets/img/integrate/quick-start-guide/create-api-from-healthcare-service.png"><img src="{{base_path}}/assets/img/integrate/quick-start-guide/create-api-from-healthcare-service.png" width="800" alt="create api dialog box"></a>

3.  Update the API name, context, and version if required, and click **Create API**. 

    The overview page of the API that you just created appears. 

    <a href="{{base_path}}/assets/img/integrate/quick-start-guide/new-healthcareapi-view.png"><img src="{{base_path}}/assets/img/integrate/quick-start-guide/new-healthcareapi-view.png" alt="apis list"></a>

4.  Navigate to **Develop -> API Configurations -> Endpoints** from the left menu. You will see that **HTTP/REST Endpoint** is already selected and the production endpoint is already provided .

>      Select the `Sandbox Endpoint`, add the endpoint `http://localhost:8290/healthcare` and **Save**.

Reviewer: ```suggestion
     Select the `Sandbox Endpoint`, add the endpoint `http://localhost:8290/healthcare`, and **Save**.
```

     
5.  Update the portal configurations and API configurations as required.

     Now, you have successfully created an API using the service.

### Step 3 - Publish the managed API

1. Navigate  to **Deployments** and click **Deploy** to create a revision to deploy in the default Gateway environment. 

> 2. Navigate to **Lifecycle** and click  **Publish** to publish the API in the Gateway environment.

Reviewer: ```suggestion
2. Navigate to **Lifecycle** and click **Publish** to publish the API in the Gateway environment.
```

### Step 4 - Invoke the Managed `HealthcareAPI` via Developer Portal

1. Navigate to  the **Developer Portal** by clicking on the `View In Dev Portal` at the top menu.

    <a href="{{base_path}}/assets/img/integrate/quick-start-guide/view-in-dev-portal.png"><img src="{{base_path}}/assets/img/integrate/quick-start-guide/view-in-dev-portal.png"></a>

2. Sign in using the default username/password `admin/admin`. You will be redirected to the **APIs**.

3. Under **APIs**, you will see the published `HealthcareAPI`. Click on it to navigate to the Overview of the API.

4. Click `Try Out`. This will create a subscription to the API using `Default Application`.

    <a href="{{base_path}}/assets/img/integrate/quick-start-guide/devportal-tryout.png"><img src="{{base_path}}/assets/img/integrate/quick-start-guide/devportal-tryout.png" size="800"></a>

5. Click `GET TEST KEY` to get a test token to invoke the API.

    <a href="{{base_path}}/assets/img/integrate/quick-start-guide/devportal-get-test-key.png"><img src="{{base_path}}/assets/img/integrate/quick-start-guide/devportal-get-test-key.png" size="800"></a>

> 6. Click **GET** resource `/doctor​/{doctorType}`. Use  **Ophthalmologist** as the doctorType. And Execute.

Reviewer: ```suggestion
6. Click **GET** resource `/doctor​/{doctorType}`. Use  **Ophthalmologist** as the doctorType and click **Execute**.
```

Reviewer: Is "Execute" clickable? If so please commit this.

Author: fixed.

        ```

2. Update and start the Micro Integrator runtime:

    1. Stop the Micro Integrator.

    2. Add the following configuration to the `<MI_HOME>/conf/deployment.toml` file of the Micro Integrator.

        !!! Tip
            The default username and password for connecting to the API gateway is `admin:admin`.

        ```toml
        [[service_catalog]]
        apim_host = "https://localhost:9443"
        enable = true
        username = "admin"
        password = "admin"
        ```
    
    3.  Start the Micro Integrator again.

        You will see the following in the server start-up log.

        ```bash
        Successfully updated the service catalog
        ```

3. Access the integration service from the **API Publisher**:

    1. Sign in to the **API Publisher**: `https://localhost:9443/publisher`

        !!! Tip
            Use `admin` as the user name and password.

    2. Select the **Services** from the menu.
    
        <img src="{{base_path}}/assets/img/integrate/quick-start-guide/select-services.png" width="500">
        
    3. See that the `HealthcareAPI` is listed as a service.

### Step 4 - Create a managed API using the Integration Service

> 

Reviewer: Remove the extra line here.

    <a href="{{base_path}}/assets/img/integrate/quick-start-guide/create-api-from-healthcare-service.png"><img src="{{base_path}}/assets/img/integrate/quick-start-guide/create-api-from-healthcare-service.png" width="800" alt="create api dialog box"></a>

3.  Update the API name, context, and version if required, and click **Create API**. 

    The overview page of the API that you just created appears. 

    <a href="{{base_path}}/assets/img/integrate/quick-start-guide/new-healthcareapi-view.png"><img src="{{base_path}}/assets/img/integrate/quick-start-guide/new-healthcareapi-view.png" alt="apis list"></a>

4.  Navigate to **Develop -> API Configurations -> Endpoints** from the left menu. You will see that **HTTP/REST Endpoint** is already selected and the production endpoint is already provided .

     Select the `Sandbox Endpoint`, add the endpoint `http://localhost:8290/healthcare` and **Save**.
     
5.  Update the portal configurations and API configurations as required.

     Now, you have successfully created an API using the service.

### Step 5 - Publish the managed API

> 1. Goto **Deployments** and Click **Deploy** to create a revision to deploy in the default gateway environment. 

Reviewer: ```suggestion
1. Navigate to **Deployments** and click **Deploy** to create a revision to deploy in the default gateway environment. 
```

Author: fixed

> 2. Goto **Lifecycle** and Click  **Publish** to publish the API in the gateway environment.

Reviewer: ```suggestion
2. Navigate to **Lifecycle** and click  **Publish** to publish the API in the Gateway environment.
```

Author: fixed

    <a href="{{base_path}}/assets/img/integrate/quick-start-guide/healthcareapi-lifecycle.png"><img src="{{base_path}}/assets/img/integrate/quick-start-guide/healthcareapi-lifecycle.png"></a>
    
    If the API is published successfully, the lifecycle state will shift to **PUBLISHED**. 

### Step 6 - Invoke the Managed `HealthcareAPI` via Developer Portal

> 1. Goto **Developer Portal** by clicking on the `View In Dev Portal` at the top menu.

Reviewer: ```suggestion
1. Navigate to the **Developer Portal** by clicking on `View In Dev Portal` at the top menu.
```

Author: fixed

    <a href="{{base_path}}/assets/img/integrate/quick-start-guide/view-in-dev-portal.png"><img src="{{base_path}}/assets/img/integrate/quick-start-guide/view-in-dev-portal.png"></a>

> 2. Sign In using the default username/pssword `admin/admin`

Reviewer: ```suggestion
2. Sign in using the default username/pssword `admin/admin`
```

Author: fixed

> 3. Go to the Overview of the `HealthcareAPI`.

Reviewer: Let's add how we get there.

Author: fixed
